opsdroid/logging.py

```diff
@@ -0,0 +1,139 @@
+"""Class for Filter logs and logging logic."""
+
+import os
+import logging
+import contextlib
+
+from opsdroid.const import DEFAULT_LOG_FILENAME, __version__
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ParsingFilter(logging.Filter):
+    """Class that filters logs."""
+
+    def __init__(self, config, *parse_list):
+        """Create object to implement filtering."""
+        super(ParsingFilter, self).__init__()
+        self.config = config["logging"]
+        self.parse_list = parse_list[0].get("whitelist") or parse_list[0].get(
```

Comment:
If the user configures both the whitelist and blacklist the blacklist will be ignored here. I could imagine this happening if you want to only see messages from one place, but it has a noisy subclass.

Either the logic needs updating to handle this. Or a warning should be raised and the docs should reflect that only one at a time can be used.

Happy either way.

Reply:
Good catch! Initially, I was thinking to allow both filters on the config, but I quickly realised that this wouldn't make much sense because when you whitelist something like - `opsdroid.core` nothing else other than the logs in `opsdroid.core` will be shown, so it wouldn't make sense to use blacklist with it.

Unless you want to mix the extended to blacklist a specific thing, but yeah not sure if this will work (i don't think you can whitelist opsdroid.core and then blacklist some function name with extended mode)

I'll update the docs and raise a warning.

Reply:
I have just tested opsdroid with both whitelist and blacklist and it breaks everything badly haha